### PR TITLE
Added code coverage to services

### PR DESCRIPTION
**Pull request by macgrewal**
Code coverage for new µ-service repos has been increased to 100% using unit/integration tests.

For backend microservice repos, `AppConfig.scala` has been trimmed to only expose a single member (to justify its existence) and a reference to it has been included in `MicroserviceHelloWorldController.scala` to achieve coverage of the code.  After discussions with other Centre Tech Leads the decision was made to make superficial use of `AppConfig.scala` as its a commonly used class and removal would likely cause rumblings in service teams.

### CODE REVIEW

**Review by beny23 — DISMISSED** [review]

File: templates/service/test/controllers/MicroserviceHelloWorldControllerSpec.scala [hunk 1]
```diff
@@ -2,14 +2,25 @@ package uk.gov.hmrc.$!APP_PACKAGE_NAME!$.controllers
...
+    new GuiceApplicationBuilder()
+      .configure(
+        "metrics.jvm"     -> false,
+        "metrics.enabled" -> false
+      )
+      .build()
 
   private val fakeRequest = FakeRequest("GET", "/")
-  private val controller = new MicroserviceHelloWorldController(Helpers.stubControllerComponents())
+  private val controller = app.injector.instanceOf[MicroserviceHelloWorldController]
```
Comment (colin-lamed) [inline templates/service/test/controllers/MicroserviceHelloWorldControllerSpec.scala:23]: I think loading the whole app to test one component is not a practice we want to promote - it makes the tests a lot slower, and seems to be done here in order to test other non-associated parts of the system (e.g. does the app start) and increase code coverage.
I think this test should stay as it is, `ErrorHandlerSpec` should also be updated to only create the `ErrorHandler`.
We should create a new test, which is explicit in it's intention, that it exists to test the app starts as a whole - e.g. `MicroserviceIntegSpec`

Comment (beny23) [reply templates/service/test/controllers/MicroserviceHelloWorldControllerSpec.scala:23]: +1

Comment (macgrewal) [reply templates/service/test/controllers/MicroserviceHelloWorldControllerSpec.scala:23]: Hi @colin-lamed , the ErrorHandler has a dependency on the error template which in turn has a dependency on the layout which has a dependency on AppConfig. Testing any controller or view purely through unit tests without injecting instances using a fake application seems unnecessarily difficult, hence the use of an application.

I totally agree with not starting up an app for tests when not needed. If this were a model, service or connector I'd typically mock things out but here the application is needed.

Comment (macgrewal) [reply templates/service/test/controllers/MicroserviceHelloWorldControllerSpec.scala:23]: WRT `MicroserviceIntegSpec`, I'll add the test and revert the changes to `MicroserviceHelloWorldController` and the associated test.

**Review by colin-lamed — APPROVED** [review]

**Pull request merged**